### ISSUE TITLE
Long term statistics not working

### ISSUE THREAD

**Issue by iotola** [tommyjlong/SpeedTest-CLI-With-Home-Assistant#4]
This verion works much better than the offical one but the long term statistics do not work in this version.
This can be fixed by including the device class and state class.
I updated my version of your python code to add these in two places for the  HAPost function call:

The definition:

`def HAPost(sensorname,state,unitmeasure,friendlyname,icon, deviceclass , stateclass):
    """Method handling sending POST to home assistant api"""
    url = HASERVER + '/api/states/sensor.' + 'speedtest_' + sensorname
    headers = {'Authorization': 'Bearer ' + AUTHKEY,
               'content-type': 'application/json'}
    attributes = {"unit_of_measurement":unitmeasure,
                  "friendly_name":friendlyname,
                  "icon":icon,
                  "device_class": deviceclass,
                  "state_class":stateclass}
    data = {'state':state, 'attributes':attributes}`

and where is it called:

`
ret1 = HAPost("download",down_load_speed,"Mbit/s","Speedtest Download","mdi:speedometer", "data_rate","measurement")
ret2 = HAPost("upload",up_load_speed,"Mbit/s","Speedtest Upload","mdi:speedometer","data_rate","measurement")
ret3 = HAPost("ping",ping_latency,"ms","Speedtest Ping","mdi:speedometer","data_rate","measurement")
`
Not sure why the code is on one line - it is not like that before its formatted!

**Comment by tommyjlong** [comment]
Thanks for the info....
Out of curiosity, how do you use the Long Term Statistics with Speednet?  
As I understand it, the History Graph uses LTS when going beyond the default 10 days, and the Statistics Lovelace Card makes use of LTS.... so just wondering how you used LTS.

**Comment by iotola** [comment]
Yes, I use the statistics card to monitor the long-term trends in bandwidth and ping, useful to see if you have introduced any problems with changes to your router setup (I use OpenWRT so tweak a lot) and I believe it would show if the ISP is slowly getting overloaded with user contention, although not run it long enough to prove that yet.
The instantaneous values are also a bit too prone to user activity.

**Comment by tommyjlong** [comment]
I made changes to the Python code so that when it reports to HA, it now includes (hopefully) **all the same attributes** that are used by the native HA speedtest.net Integration, which includes the device class and state class.  Confirmed the device/state class settings will allow Long Term Statistics hourly events to be created in the Statistics Table.  For anyone not wanting LTS, there is a configuration option to not report the device/state class in which case, LTS data is not created in the Statistics Table.
